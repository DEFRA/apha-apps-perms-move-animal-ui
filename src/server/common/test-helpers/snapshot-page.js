import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
