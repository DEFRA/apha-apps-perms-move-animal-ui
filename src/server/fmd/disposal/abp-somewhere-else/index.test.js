import { describePageSnapshot } from '../../../common/test-helpers/snapshot-page.js'
import { AbpSomewhereElseExitPage } from './index.js'

const pageUrl = '/fmd/disposal-of-animals/ABP-somewhere-else'

describe('#DisposalExitPage', () => {
  it('should have the correct properties', () => {
    const page = new AbpSomewhereElseExitPage()
    expect(page.pageTitle).toBe(
      'You must dispose of animal by products at certain locations'
    )
    expect(page.urlPath).toBe(pageUrl)
    expect(page.view).toBe('fmd/disposal/abp-somewhere-else/index')
  })

  describePageSnapshot({
    describes: 'DisposalExitPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
