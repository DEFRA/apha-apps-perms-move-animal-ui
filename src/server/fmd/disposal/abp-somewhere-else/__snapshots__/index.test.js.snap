// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#DisposalExitPage DisposalExitPage.content should render expected response and content 1`] = `
"
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <h1 class="govuk-heading-l">You must dispose of animal by products at certain locations</h1>
      <div class="govuk-body" data-testid="app-page-body">
        <p>They can only move to one of the following places:</p>
        <ul class="govuk-list govuk-list--bullet">
          <li>Rendering plant</li>
          <li>Incinerator</li>
        </ul>
      </div>
    </div>
  </div>
        "
`;
